packages/@aws-cdk/aws-appmesh/lib/virtual-service.ts

```diff
@@ -50,19 +61,75 @@ export interface VirtualServiceProps {
   readonly virtualServiceProvider: VirtualServiceProvider;
 }
 
+abstract class VirtualServiceBase extends cdk.Resource implements IVirtualService {
+  /**
+   * The name of the VirtualService
+   *
+   * @attribute
+   */
+  public abstract readonly virtualServiceName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) for the virtual service
+   *
+   * @attribute
+   */
+  public abstract readonly virtualServiceArn: string;
+
+  /**
+   * The Mesh which the VirtualService belongs to
+   */
+  public abstract readonly mesh: IMesh;
+
+  /**
+   * Grants the given entity all read permissions for this VirtualService.
+   */
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:DescribeVirtualService',
+      'appmesh:ListVirtualService',
+      'appmesh:ListTagsForResource',
+    );
+  }
+
+  /**
+   * Grants the given entity all write permissions for this VirtualService.
+   */
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:CreateVirtualService',
+      'appmesh:UpdateVirtualService',
+      'appmesh:DeleteVirtualService',
+      'appmesh:TagResource',
+      'appmesh:UntagResource',
```

Comment:
Same questions here as for `GatewayRoute` - what would necessitate a resource having these permissions?